crates/ruff_python_parser/src/string.rs

```diff
@@ -629,61 +357,94 @@ pub(crate) fn parse_strings(
                 implicit_concatenated,
             }),
             kind: initial_kind,
-            range: TextRange::new(initial_start, last_end),
+            range,
         }
         .into());
     }
 
     // De-duplicate adjacent constants.
     let mut deduped: Vec<Expr> = vec![];
     let mut current: Vec<String> = vec![];
```

Comment:
Should we change this to a `String` and use `push_str` directly instead of pushing parts and joining them at the end?

Reply:
Yes, we can then use `std::mem::take` to move the `String`.